test/ServiceRegistry.js

```diff
@@ -84,6 +85,11 @@ describe("ServiceRegistry", function () {
         serviceRegistryL2 = await ServiceRegistryL2.deploy("Service Registry L2", "SERVICE", "https://localhost/service/");
         await serviceRegistryL2.deployed();
 
+        const ServiceRegistryAnnotated = await ethers.getContractFactory("ServiceRegistryAnnotated");
+        serviceRegistryAnnotated = await ServiceRegistryAnnotated.deploy("Service Registry Annotated", "SERVICE",
+            "https://localhost/service/", agentRegistry.address);
+        await serviceRegistryAnnotated.deployed();
```

Comment:
Adding an annotated registry testing. Note that the contract needs to be instrumented before running this test, otherwise the contract is not going to be different from the original ServiceRegistry contract.